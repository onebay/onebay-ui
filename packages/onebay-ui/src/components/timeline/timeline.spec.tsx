import { mount } from '@vue/test-utils'
import Timeline from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(Timeline, {
    slots,
    props: { ...values }
  })
}

describe('Timeline Snap', () => {
  it('render initial Timeline', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Timeline -- props style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Timeline -- props pending', () => {
    const wrapper = factory()
    wrapper.setProps({ pending: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Timeline -- props items', () => {
    const wrapper = factory({
      items: [
        { title: 'Eat breakfast', content: ['8:00'] },
        { title: 'Go to work', color: 'yellow' },
        { title: 'Sleep', icon: 'clock' }
      ]
    })
    expect(wrapper.element).toMatchSnapshot()
  })

})
