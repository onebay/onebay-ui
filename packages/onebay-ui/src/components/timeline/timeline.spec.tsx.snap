// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline Snap render Timeline -- props customStyle 1`] = `
<div
  class="ob-timeline"
  style="color: red;"
>
  
  
</div>
`;

exports[`Timeline Snap render Timeline -- props items 1`] = `
<div
  class="ob-timeline"
>
  
  <div
    class="ob-timeline-item"
  >
    <div
      class="ob-timeline-item__tail"
    />
    <div
      class="ob-timeline-item__dot"
    >
      <!---->
    </div>
    <div
      class="ob-timeline-item__content"
    >
      <div
        class="ob-timeline-item__content-item"
      >
        Eat breakfast
      </div>
      
      <div
        class="ob-timeline-item__content-item ob-timeline-item__content--sub"
      >
        8:00
      </div>
      
    </div>
  </div>
  <div
    class="ob-timeline-item ob-timeline-item--yellow"
  >
    <div
      class="ob-timeline-item__tail"
    />
    <div
      class="ob-timeline-item__dot"
    >
      <!---->
    </div>
    <div
      class="ob-timeline-item__content"
    >
      <div
        class="ob-timeline-item__content-item"
      >
        Go to work
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ob-timeline-item"
  >
    <div
      class="ob-timeline-item__tail"
    />
    <div
      class="ob-timeline-item__icon"
    >
      <div
        class="ob-icon ob-icon-clock"
      />
    </div>
    <div
      class="ob-timeline-item__content"
    >
      <div
        class="ob-timeline-item__content-item"
      >
        Sleep
      </div>
      <!---->
    </div>
  </div>
  
</div>
`;

exports[`Timeline Snap render Timeline -- props pending 1`] = `
<div
  class="ob-timeline"
>
  
  
</div>
`;

exports[`Timeline Snap render initial Timeline 1`] = `
<div
  class="ob-timeline"
>
  
  
</div>
`;
